### PR TITLE
Memory allocation, use std::vector 

### LINKED ISSUE #1390: Memory allocation, use std::vector
MemAlloc is still used in those files, replace it with std::vector
- [ ] simulation/sim_allocation_tableaux.cpp
- [ ] optimisation/opt_alloc_probleme_a_optimiser.cpp
- [ ] optimisation/opt_nombre_min_groupes_demarres_couts_demarrage.cpp

### PR DESCRIPTION
**Pull request by payetvin**
- [x] simulation/sim_allocation_tableaux.cpp
- [ ] optimisation/opt_alloc_probleme_a_optimiser.cpp
- [ ] optimisation/opt_nombre_min_groupes_demarres_couts_demarrage.cpp

### CODE REVIEW

**Review by flomnes — DISMISSED** [review]
`TransitMoyenRecalculQuadratique` is no longer initialized

File: src/solver/hydro/management/daily.cpp [hunk 1]
```diff
@@ -237,7 +237,7 @@ inline void HydroManagement::prepareDailyOptimalGenerations(Solver::Variable::St
     auto const& maxP = maxPower[Data::PartHydro::genMaxP];
     auto const& maxE = maxPower[Data::PartHydro::genMaxE];
 
-    auto const& valgen = *ValeursGenereesParPays[numSpace][z];
+    auto& valgen = ValeursGenereesParPays[numSpace][z];
```
Comment (JasonMarechal25) [inline src/solver/hydro/management/daily.cpp:240]: _Const_ vanished

File: src/solver/simulation/sim_allocation_tableaux.cpp [hunk 2]
```diff
@@ -38,163 +38,52 @@ using namespace Antares;
...
-        rpNtc->VariablesDualesMoyennes = (double*)MemAlloc(sizeOfDoubleHours);
-        rpNtc->RenteHoraire = (double*)MemAlloc(sizeOfDoubleHours);
-    }
-}
-
-static void DeallocateResultsForEconomicMode(void)
-{
-    auto& study = *Data::Study::Current::Get();
-    RESULTATS_PAR_INTERCONNEXION* rpNtc;
+    ResultatsParInterconnexion.resize(1 + study.runtime->interconnectionsCount());
```
Comment (flomnes) [inline src/solver/simulation/sim_allocation_tableaux.cpp]: I don't think the +1 is required here, you should try to remove it

File: src/solver/simulation/sim_structure_donnees.h [hunk 3]
```diff
@@ -70,16 +61,7 @@ typedef struct
...
-    int* TransitMinimumNo;  // UNUSED
-    double* TransitMaximum; // UNUSED
-    int* TransitMaximumNo;  // UNUSED
-    double* TransitStdDev;  // UNUSED
-    double* TransitMoyenRecalculQuadratique;
-    double* VariablesDualesMoyennes; // UNUSED
-    double* TransitAnnuel;           // UNUSED
-    double* RenteHoraire;            // UNUSED
+    std::vector<double> TransitMoyenRecalculQuadratique;
 } RESULTATS_PAR_INTERCONNEXION;
```
Comment (JasonMarechal25) [inline src/solver/simulation/sim_structure_donnees.h]: Is the struct leftover necessary ?
Can't we just have `std::vector<double> RESULTATS_PAR_INTERCONNEXION` ou `std::vector<double> TransitMoyenRecalculQuadratique`

**Review by JasonMarechal25 — COMMENTED** [review]
Some comments

File: src/solver/hydro/management/daily.cpp (hunk 1 above)
Comment (payetvin) [reply src/solver/hydro/management/daily.cpp:240]: `valgen.HydrauliqueModulableQuotidien[day] = problem.Turbine[dayMonth];`
It's modified later in the same file, I don't know why it was declared const in the first place

File: src/solver/simulation/sim_structure_donnees.h (hunk 3 above)
Comment (payetvin) [reply src/solver/simulation/sim_structure_donnees.h]: We could have a simple std::vector<std::vector<double>> but I think keep the struct name is clearer, @flomnes what do you think ?

Comment (flomnes) [reply src/solver/simulation/sim_structure_donnees.h]: No, you could keep only TransitMoyen**Interconnexions**RecalculQuadratique

File: src/solver/hydro/management/daily.cpp (hunk 1 above)
Comment (JasonMarechal25) [reply src/solver/hydro/management/daily.cpp:240]: Maybe the pointer was const, not the value

**Review by JasonMarechal25 — APPROVED** [review]

**Pull request merged**